### DISCUSSION TITLE
Any news on updating this library for the Teensy 4 boards?

### DISCUSSION THREAD

**Discussion by luni64** [luni64/TeensyStep#105] [General]
I have a working version but it is not yet stable enough to publish. I'll add the most requested features, i.e.,

- ability to change targets on the fly,
- support more than 4 independently moving motors

I also noticed that a lot of users have difficulties understanding the controller concept. So, I'll redo the API to make moving motors and group of motors more intuitive.

I have no schedule at the moment since I'm still involved in some higher priority projects.

**Comment by luni64** [comment]
Here some examples of the new API
```c++
Stepper s_x(0, 1), s_y(2, 3), s_z(4, 5);

StepperGroup all{s_x, s_y, s_z};      // group of all motors

void someFunc()
{   
    s_z.moveAbs(1000);                //move a stepper directly

    s_x.setTargetAbs(200);            // define targets for s_x ...
    s_y.setTargetAbs(500);            // ... and s_y
    StepperGroup xy{s_x, s_y};        // define a group of steppers
    xy.move();                        // move the group synchronously

     for (Stepper s : {s_x, s_y, s_z}) // set all targets to 0
    {
        s.setTargetAbs(0);
    }

    all.move();                       // move all motors home
}

```

**Comment by dbartelmus** [comment]
Is it possible now to use/test it on T4.x boards?

**Comment by luni64** [comment]
I didn't find a lot of time lately to work on this library. You can however give it a try. I gave you access to the (private) repository (should have  gotten a mail from gitHub). Please do not push to this repository. There probably are a lot of bugs in it but it was good enough for this:  https://www.youtube.com/watch?v=zX35L4OJa68&t=15s

Here the firmware for the dispenser, might be useful as example:  [DispControl.zip](https://github.com/luni64/TeensyStep/files/6803126/DispControl.zip)

**Comment by dbartelmus** [comment]
Hey, thanks.
I will take a look at this with 4 ac servo motor setup

W dniu pon., 12.07.2021 o 18:45 luni64 ***@***.***>
napisał(a):

> I didn't find a lot of time lately to work on this library. You can
> however give it a try. I gave you access to the (private) repository
> (should have gotten a mail from gitHub). Please do not push to this
> repository. There probably are a lot of bugs in it but it was good enough
> for this: https://www.youtube.com/watch?v=zX35L4OJa68&t=15s
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/luni64/TeensyStep/discussions/105#discussioncomment-994659>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABPTVEZXXSGPZFER4RE5WHLTXML35ANCNFSM45AUAK4Q>
> .
>
-- 
DAMIAN BARTELMUSIT Specialist / DevOps

www.panther.software <https://panther.software/>
+48-PANTHER-00 (+48726843700)

**Comment by luni64** [comment]
Just did a quick test. 
Sorry, looks like the version on GitHub is not the latest and is not functional. Unfortunately I'm currently not at home and can't upload a more recent version. I'll push it on Wednesday

**Comment by dbartelmus** [comment]
Thank you, im waiting then.

W dniu pon., 12.07.2021 o 19:33 luni64 ***@***.***>
napisał(a):

> Just did a quick test.
> Sorry, looks like the version on GitHub is not the latest and is not
> functional. Unfortunately I'm currently not at home and can't upload a more
> recent version. I'll push it on Wednesday
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/luni64/TeensyStep/discussions/105#discussioncomment-994933>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABPTVE6NU33LTR4RTUXYXHLTXMROZANCNFSM45AUAK4Q>
> .
>
-- 
DAMIAN BARTELMUSIT Specialist / DevOps

www.panther.software <https://panther.software/>
+48-PANTHER-00 (+48726843700)

**Reply by ddhille** [reply]
waiting desperately for a teensy 4.1 version 🤩
does anyone have some workaround?
or whats your status here?

**Reply by AlexHasker** [reply]
Me too, the Library looks so promising

**Comment by defenestrated** [comment]
@luni64 another vote for an update! your library looks like the magic bullet to fix a gnarly project i'm working on ...but i need it to work with 4.1... any way you could publish the working version, even if it's not totally stable?

**Comment by luni64** [comment]
I still have to apologize that I still don't find much time to finalize the T4 version.  Anyway, I published the current state in this repo: https://github.com/luni64/TeensyStep4 for first trials. The examples section contains two usage examples. Here another one: https://github.com/luni64/TeensyStep4/discussions/2

Feedback in the discussion or issues section is very welcome, But, please note that most likely I won't be able to fix bugs or give extended support at the moment. 

Hope that helps

**Reply by defenestrated** [reply]
Totally understood – thank you for your work on this and for posting what you have! This'll be a great starting point, and maybe some of us can help triage bugs if they emerge. Thanks again =)
